apps/music/js/metadata/album_art_cache.js

```diff
@@ -164,12 +227,13 @@ var AlbumArtCache = (function() {
    *
    * @param {String} cacheKey The cache key generated by makeCacheKey.
    * @param {Blob} blob The blob.
+   * @param {String} type Either 'thumbnail' or 'fullsize'.
    * @return {String} The blob: URL.
    */
-  function makeURL(cacheKey, blob) {
+  function makeURL(cacheKey, blob, type) {
     var url = URL.createObjectURL(blob);
     if (cacheKey) {
-      L1Cache[cacheKey] = url;
+      L1Cache[type][cacheKey] = url;
```

Comment:
Before creating this new URL you could make sure that the cache is still empty. If not, return the cached URL instead of creating the new one.

Or, if you're not sure that would be safe, you could at least emit a console.log here to let you know that your race condition had occurred. That would give you some data about how common it is in practice.